### PR TITLE
Made the client swarm-compatible

### PR DESCRIPTION
**Pull request by xFragger**
1. InfoCmd sends info about the SwarmNodes, had to parse that
2. The Containers-Names array is a bit confusing, but i managed to clean it up + get the swarmhostname that container is running on

Tested what i was able to test. Didn't manage to create a swarm inside the test-suite, so an integration test of all this is missing atm.

Please discuss, i will make all suggested and useful changes, in 2 weeks at the latest.

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/443)

### CODE REVIEW

**Comment by xFragger** [comment]
oh, i just saw the other pull requests... sorry for duplicated stuff.... but all i wrote, was written weeks ago, i just wanted to give back stuff

**Comment by xFragger** [comment]
another thing => if i would be able to retrieve the value of isSwarmEndpoint inside the PullImageResultItem, the check for " downloaded" could be better

also the "Cast Info to SwarmInfo" thing could be better i guess

**Comment by sray7965** [comment]
By when do you guys plan to release Snapshot 3.0.0 into maven.

**Comment by marcuslinke** [comment]
@xFragger What we need for swarm related changes is an integration test case. See https://github.com/docker-java/docker-java/pull/337 please for a proposal. I've currently no time for implementing a setup as described there. Feel free to provide an PR with an integration test setup and we can start to merge swarm related things into master. @KostyaSha : Do you agree?

**Comment by marcuslinke** [comment]
@sray7965 I'm planning to release v3.0.0 until 15th of february.

**Comment by KostyaSha** [comment]
@marcuslinke i would prefer to split integration tests into separate module and reorganise to multi-module before.

**Comment by marcuslinke** [comment]
@KostyaSha Makes sense. :+1:

**Comment by marcuslinke** [comment]
Related to issue https://github.com/docker-java/docker-java/issues/395

File: src/main/java/com/github/dockerjava/api/model/Container.java [hunk 1]
```diff
@@ -42,6 +42,11 @@
     @JsonProperty("Status")
     private String status;
 
+    @JsonIgnore
+    private String name;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/Container.java]: if it temp, then should be transient

File: src/main/java/com/github/dockerjava/api/model/PullResponseItem.java [hunk 2]
```diff
@@ -11,13 +11,13 @@
...
-    private static final String IMAGE_UP_TO_DATE = "Image is up to date";
-
-    private static final String DOWNLOAD_COMPLETE = "Download complete";
+    private static final String[] SUCCESS_STRINGS = new String[]{
+            "this image was pulled from a legacy registry",
+            "Downloaded newer image",
+            "Image is up to date",
+            "Download complete",
+            " downloaded"                   //Swarm sends another text, when download is complete!
+    };
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/PullResponseItem.java]: May complicate testing against single messages.

Comment (xFragger) [inline src/main/java/com/github/dockerjava/api/model/PullResponseItem.java]: "may"? so should revert this and just add the new one or leave it?

**Comment by xFragger** [comment]
Well.... look like it works... but one problem => had to hardcode the TLS Stuff, any ideas how to do this better?

File: src/main/java/com/github/dockerjava/api/model/PullResponseItem.java [hunk 3]
```diff
@@ -10,14 +10,12 @@
...
     private static final long serialVersionUID = -2575482839766823293L;
-
     private static final String LEGACY_REGISTRY = "this image was pulled from a legacy registry";
-
     private static final String DOWNLOADED_NEWER_IMAGE = "Downloaded newer image";
-
     private static final String IMAGE_UP_TO_DATE = "Image is up to date";
-
     private static final String DOWNLOAD_COMPLETE = "Download complete";
+    private static final String SWARM_DOWNLOADED = " downloaded"; //Swarm sends another text, when download is complete!
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/PullResponseItem.java]: remove comment

File: src/main/java/com/github/dockerjava/api/model/SwarmInfo.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            if (nodesFound && !key.substring(0, 3).equals(" └ ")) {
+                Node node = new Node(key, lineArr.get(1));
+                nodes.put(key, node);
+                lastNode = node;
+                continue;
+            }
+
+            if (lastNode != null) {
+                switch (key) {
+                    case " └ Status":
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/SwarmInfo.java]: looks terrible... is it API answer?

Comment (xFragger) [inline src/main/java/com/github/dockerjava/api/model/SwarmInfo.java]: it IS terrible... but the api answers with an backspace before these  └  signs....

**Comment by KostyaSha** [comment]
Could you check whether master contains this functionality?

**Pull request closed without merging**